Source/WebCore/editing/cocoa/DataDetection.h

```diff
@@ -33,7 +33,11 @@
 #import <wtf/OptionSet.h>
 #import <wtf/RetainPtr.h>
 
+#if HAVE(SECURE_ACTION_CONTEXT)
+OBJC_CLASS DDSecureActionContext;
+#else
 OBJC_CLASS DDActionContext;
+#endif
```

Comment:
You should add something like this to `DataDetectorsSPI.h`:

```
#if HAVE(SECURE_ACTION_CONTEXT)
using WKDDActionContext = DDSecureActionContext;
#else
using WKDDActionContext = DDActionContext;
#endif
```

Then instead of doing the `#if HAVE(SECURE_ACTION_CONTEXT)` dance everywhere, you can just use `WKDDActionConext` instead.  (I'm hoping this works for Obj-C types in both C++ and Obj-C++ contexts, but I've never tried.)

Reply:
That works for all the ObjC.  There's only one use in a .cpp file (WKBundlePageOverlay.cpp) which is going to be removed after rdar://108925315 is resolved.

Reply:
It also doesn't work for headers, but it cleaned up all the .mm files nicely.